### ISSUE TITLE
All blank output from ctc_decoder

### ISSUE THREAD

**Issue by rbracco** [pytorch/audio#2584]
### 🐛 Describe the bug

I am following [the torchaudio ctc_decoder tutorial](https://pytorch.org/audio/main/tutorials/asr_inference_with_ctc_decoder_tutorial.html) and getting seemingly random outputs that decode to the empty string.

Here is how I instantiate and call the decoder

```python
beam_search_decoder = ctc_decoder(
                                        tokens = tokens, 
                                        lexicon=None, 
                                        nbest=3, 
                                        lm=None, 
                                        log_add=True, 
                                        beam_size=1500, 
                                        blank_token="<blank>", 
                                        sil_token=" "
                                         )
# transpose the probs to be (batch, frame, num_tokens) as stated in documentation, shape is [1, 174,56] after transpose
beam_search_result = beam_search_decoder(probs.transpose(1,2))
beam_search_transcript = " ".join(beam_search_result[0][0].words).strip()
```
`beam_search_transcript` is always the empty string. 

The non-transposed probabilities (shape [1, 56, 174]) can be downloaded [here on Google Drive](https://drive.google.com/file/d/112L112yEE3evWa-jZvLANjEH3qZzwxqa/view?usp=sharing) if needed

My tokens are these 56 chars:
```
[' ', 'aɪ', 'aʊ', 'b', 'b̚', 'd', 'd̚', 'eɪ', 'f', 'g', 'h', 'i', 'j', 'k', 'kʰ', 'k̚', 'l', 'l̩', 'm', 'm̩', 'n', 'n̩', 'oʊ', 'p', 'pʰ', 'p̚', 's', 't', 'tʰ', 't̚', 'u', 'v', 'w', 'z', 'æ', 'ð', 'ŋ', 'ɑ', 'ɔ', 'ɔɪ', 'ə', 'ɚ', 'ɛ', 'ɡ̚', 'ɪ', 'ɫ', 'ɹ', 'ɾ', 'ʃ', 'ʊ', 'ʒ', 'ʔ', 'ʤ', 'ʧ', 'θ', '<blank>']
```

I have tried `log_add=True` and `log_add=False` as discussed in #2558 and still the output is the empty string

The output of `beam_search_result[0][0]` looks like this:
![image](https://user-images.githubusercontent.com/47190785/181287632-63f261d6-d09a-4b35-b45f-be366cafed34.png)

I have successfully used both ctcdecode, and pyctcdecode in the past so I am familiar with beam searching and the various params, but I can't figure out what I'm doing wrong, or if there is a bug of some kind. 

### Versions

torch 1.12.0
torchaudio 0.12.0

**Comment by carolineechen** [comment]
Hi @rbracco, thanks for the issue. The `words` entry is only applicable when using a lexicon, in which case we store the generated word/lexicon corresponding to the generated tokens. Since you are not using a lexicon, we don't include the corresponding output `words` (but maybe it would make sense for us to include a transcription that is split by sil token in the future?). Instead, you can generate it yourself using the `idx_to_tokens` function to convert the token IDs to their corresponding tokens.

`beam_search_decoder.idx_to_tokens(beam_search_result[0][0].tokens)` will produce the list of generated tokens.

**Comment by rbracco** [comment]
Hey @carolineechen, thank you for the clarification. I have switched to using `.tokens` as suggested but the token values are still not correct. The actual text is only 46 tokens long, but the beam prediction is 475 (close to the total number of timesteps, 496). My code is the same as the above except I am now correctly referencing `.tokens` instead of `.words`, any ideas where I might be going wrong? Thanks!

Here are the argmaxes for my emissions/logits
![image](https://user-images.githubusercontent.com/47190785/182248885-639d28bb-1a6c-4fdc-bf48-d930acbb16d5.png)
and here are the tokens of the top scoring beam search results:
![image](https://user-images.githubusercontent.com/47190785/182248931-37ad1cc2-f338-49fa-bc15-618029dc0803.png)
Output of `"".join(beam_search_decoder.idxs_to_tokens(beam_search_result[0][0].tokens))`:
![image](https://user-images.githubusercontent.com/47190785/182249588-490702c8-a7a1-4d3b-a723-7b6b3debb5f9.png)

**Comment by rbracco** [comment]
I figured it out, I didn't call `torch.nn.functional.log_softmax` on the probabilities/emissions before decoding. Maybe the docs/tutorial could make this explicit as it is a common thing people trip over when doing ctc decoding? Thanks for your help @carolineechen
